### PR TITLE
pythonPackages.h5py: 2.7.1 -> 2.8.0

### PR DESCRIPTION
**Pull request by jluttine**
###### Motivation for this change

Upgrade h5py.

(I'm running the nox-review at the moment but I guess it can take a very long time.)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by lsix** [comment]
Looks good to me. Lets see how the build goes.

@GrahamcOfBorg build pythonPackages.h5py python3Packages.h5py

**Comment by jluttine** [comment]
I ran nox-review but some packages failed to build. Not sure if they were failing anyway before this change...

File: pkgs/development/python-modules/h5py/default.nix [hunk 1]
```diff
@@ -30,18 +30,13 @@ in buildPythonPackage rec {
 
   preBuild = if mpiSupport then "export CC=${mpi}/bin/mpicc" else "";
 
+  checkInputs = optional (isPy26 || isPy27) unittest2;
```
Comment (xeji) [inline pkgs/development/python-modules/h5py/default.nix]: `isPy26` test is obsolete, we don't have python26 in nixpkgs anymore.

Comment (jluttine) [reply pkgs/development/python-modules/h5py/default.nix]: Ok, removed.

**Comment by xeji** [comment]
@GrahamcOfBorg build pythonPackages.h5py python3Packages.h5py

**Comment by dotlambda** [comment]
Edward and tflearn were broken before.

**Pull request merged**